radix-engine/src/engine/interpreters/native_interpreter.rs

```diff
@@ -74,74 +75,145 @@ impl Into<ApplicationError> for EpochManagerError {
-                ResourceManager::static_main(func, input, system_api).map_err(|e| e.into())
+pub trait InvokableNative<'a>:
+    Invokable<EpochManagerCreateInvocation>
+    + Invokable<PackagePublishInvocation>
+    + Invokable<ResourceManagerBucketBurnInvocation>
+    + Invokable<ResourceManagerCreateInvocation>
+    + Invokable<TransactionProcessorRunInvocation<'a>>
+    + Invokable<BucketTakeInvocation>
+    + Invokable<BucketCreateProofInvocation>
+    + Invokable<BucketTakeNonFungiblesInvocation>
+    + Invokable<BucketGetNonFungibleIdsInvocation>
+    + Invokable<BucketGetAmountInvocation>
+    + Invokable<BucketPutInvocation>
+    + Invokable<BucketGetResourceAddressInvocation>
+    + Invokable<AuthZonePopInvocation>
+    + Invokable<AuthZonePushInvocation>
+    + Invokable<AuthZoneCreateProofInvocation>
+    + Invokable<AuthZoneCreateProofByAmountInvocation>
+    + Invokable<AuthZoneCreateProofByIdsInvocation>
+    + Invokable<AuthZoneClearInvocation>
+    + Invokable<AuthZoneDrainInvocation>
+    + Invokable<ProofGetAmountInvocation>
+    + Invokable<ProofGetNonFungibleIdsInvocation>
+    + Invokable<ProofGetResourceAddressInvocation>
+    + Invokable<ProofCloneInvocation>
+    + Invokable<WorktopPutInvocation>
+    + Invokable<WorktopTakeAmountInvocation>
+    + Invokable<WorktopTakeAllInvocation>
+    + Invokable<WorktopTakeNonFungiblesInvocation>
+    + Invokable<WorktopAssertContainsInvocation>
+    + Invokable<WorktopAssertContainsAmountInvocation>
+    + Invokable<WorktopAssertContainsNonFungiblesInvocation>
+    + Invokable<WorktopDrainInvocation>
+    + Invokable<VaultTakeInvocation>
+    + Invokable<VaultPutInvocation>
+    + Invokable<VaultLockFeeInvocation>
+    + Invokable<VaultTakeNonFungiblesInvocation>
+    + Invokable<VaultGetAmountInvocation>
+    + Invokable<VaultGetResourceAddressInvocation>
+    + Invokable<VaultGetNonFungibleIdsInvocation>
+    + Invokable<VaultCreateProofInvocation>
+    + Invokable<VaultCreateProofByAmountInvocation>
+    + Invokable<VaultCreateProofByIdsInvocation>
+    + Invokable<ComponentAddAccessCheckInvocation>
+    + Invokable<ResourceManagerBurnInvocation>
+    + Invokable<ResourceManagerUpdateAuthInvocation>
+    + Invokable<ResourceManagerLockAuthInvocation>
+    + Invokable<ResourceManagerCreateVaultInvocation>
+    + Invokable<ResourceManagerCreateBucketInvocation>
+    + Invokable<ResourceManagerMintInvocation>
+    + Invokable<ResourceManagerGetMetadataInvocation>
+    + Invokable<ResourceManagerGetResourceTypeInvocation>
+    + Invokable<ResourceManagerGetTotalSupplyInvocation>
+    + Invokable<ResourceManagerUpdateMetadataInvocation>
+    + Invokable<ResourceManagerUpdateNonFungibleDataInvocation>
+    + Invokable<ResourceManagerNonFungibleExistsInvocation>
+    + Invokable<ResourceManagerGetNonFungibleInvocation>
+    + Invokable<ResourceManagerSetResourceAddressInvocation>
+    + Invokable<EpochManagerGetCurrentEpochInvocation>
+    + Invokable<EpochManagerSetEpochInvocation>
```

Comment:
I think we will need more thoughts on forks very soon. IE if we add more invokable in the future, we will need combination of invokables.

Reply:
I was thinking what would happen if we've added extra invocation and we need to run past transaction without the added invocations.

Reply:
ah, gotcha. Good point! I think one easy approach may be separate traits for each fork version?